src/internal/obj/monkey_client.go

```diff
@@ -119,20 +79,8 @@ func (c *monkeyClient) Walk(ctx context.Context, dir string, walkFn func(name st
 
 // Exists wraps the existance check.
 func (c *monkeyClient) Exists(ctx context.Context, path string) bool {
+	if enabled && localRand.Float64() < failProb {
```

Comment:
The monkey client isn't supposed to give the wrong results, just to occasionally fail the retryable operations (putting / getting).

Reply:
Failure is equivalent to false here, so we should test for it.

Reply:
Now that we are returning an error, we should be good. We need to avoid giving incorrect results and reporting no errors, because that is not the correct semantics for the monkey client.